code/frameworks/nextjs/README.md

```diff
@@ -0,0 +1,622 @@
+
+For example, consider the following Next.js config:
+
+```js
+// next.config.js
+module.exports = {
+  serverRuntimeConfig: {
+    mySecret: 'secret',
+    secondSecret: process.env.SECOND_SECRET, // Pass through env variables
+  },
+  publicRuntimeConfig: {
+    staticFolder: '/static',
+  },
+};
+```
+
+Calls to `getConfig` would return the following object when called within Storybook:
+
+```json
+{
+  "serverRuntimeConfig": {},
+  "publicRuntimeConfig": {
+    "staticFolder": "/static"
+  }
+}
+```
+
+### Custom Webpack Config
+
+Next.js comes with a lot of things for free out of the box like sass support, but sometimes you add [custom webpack config modifications to Next.js](https://nextjs.org/docs/api-reference/next.config.js/custom-webpack-config). This framework takes care of most of the webpack modifications you would want to add. If Next.js supports a feature out of the box, then that feature will work out of the box in Storybook. If Next.js doesn't support something out of the box, but makes it easy to configure, then this framework will do the same for that thing for Storybook.
+
+Any webpack modifications desired for Storybook should be made in [.storybook/main.js](https://storybook.js.org/docs/react/builders/webpack#extending-storybooks-webpack-config).
+
+Note: Not all webpack modifications are copy/paste-able between `next.config.js` and `.storybook/main.js`. It is recommended to do your reasearch on how to properly make your modifcation to Storybook's webpack config and on how [webpack works](https://webpack.js.org/concepts/).
+
+Below is an example of how to add svgr support to Storybook with this framework.
+
+```js
+// .storybook/main.js
+module.exports = {
+  // other config omitted for brevity
+  webpackFinal: async (config) => {
+    // this modifies the existing image rule to exclude .svg files
+    // since you want to handle those files with @svgr/webpack
+    const imageRule = config.module.rules.find((rule) => rule.test.test('.svg'));
+    imageRule.exclude = /\.svg$/;
+
+    // configure .svg files to be loaded with @svgr/webpack
+    config.module.rules.push({
+      test: /\.svg$/,
+      use: ['@svgr/webpack'],
+    });
+
+    return config;
+  },
+};
+```
+
+### Typescript
+
+Storybook handles most [Typescript](https://www.typescriptlang.org/) configurations, but this framework adds additional support for Next.js's support for [Absolute Imports and Module path aliases](https://nextjs.org/docs/advanced-features/module-path-aliases). In short, it takes into account your `tsconfig.json`'s [baseUrl](https://www.typescriptlang.org/tsconfig#baseUrl) and [paths](https://www.typescriptlang.org/tsconfig#paths). Thus, a `tsconfig.json` like the one below would work out of the box.
+
+```json
+{
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "@/components/*": ["components/*"]
+    }
+  }
+}
+```
+
+### next.config.js
+
+#### ESM
```

Comment:
Is this still true now that this is a framework?

Reply:
Hey @RyanClementsHax ! I added support for esm in [this commit](https://github.com/storybookjs/storybook/pull/19382/commits/252fb59205b3515a30076539891fb46892bd5b00), in case you're interested!

Reply:
🤯 I can't believe I forgot about 'import'!